### PR TITLE
Homepage tests fix

### LINKED ISSUE #23: add a monitor for home page content
@davidnuescheler has requested that we create a monitor that can ensure there aren't 404s on the homepage! This is a valuable tool for ensuring prominent content isn't missing.

### PR DESCRIPTION
**Pull request by MarquiseRosier**
Fix #23

### CODE REVIEW

**Comment by kptdobe** [comment]
I assume this is a PR for https://github.com/adobe/theblog-tests/issues/23...

File: .vscode/launch.json [hunk 1]
```diff
@@ -52,6 +52,19 @@
             "args":[
                 "up"
             ]
-        }
+        },
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "homepage",
+            "program": "${workspaceFolder}/monitoring/homepage_test",
```
Comment (kptdobe) [inline .vscode/launch.json:60]: I do not think this should be committed here, really specific to your work.

File: monitoring/homepage_test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const assert = require('assert');
+
+function randomArray(digits, max){
+  return Array.from({length: digits}, () => Math.round(Math.random() * max));
+}
+
+function detectFails(sitePage) {
+  return (response) => {
+    if (response.statusCode !== 200) {
+      assert.fail(`${sitePage} === ${response.statusCode}`);
```
Comment (kptdobe) [inline monitoring/homepage_test.js:27]: Nitpicking but maybe the failure description could be a bit more explicit than just "url === statusCode"

**Review by kptdobe — APPROVED** [review]

**Review by rofe — APPROVED** [review]

File: monitoring/homepage_test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/* eslint-disable no-console */
+/* global $http $util $secure */
+/*
+ * Synthetics API Test Documentation:
+ * https://docs.newrelic.com/docs/synthetics/new-relic-synthetics/scripting-monitors/write-synthetics-api-tests
+ */
+const assert = require('assert');
+
+function randomArray(digits, max){
+  return Array.from({length: digits}, () => Math.round(Math.random() * max));
```
Comment (trieloff) [inline monitoring/homepage_test.js:21]: Can we get a constant seed (maybe per location)?

Comment (MarquiseRosier) [reply monitoring/homepage_test.js]: I just removed the randomness; no real need we should just check all of the first 256 of the index especially when it doesn't really cost to do so! :)

Comment (MarquiseRosier) [reply monitoring/homepage_test.js]: Maybe the parts of the index that we check should be randomized? we check max 256 the [start + offset] can be random? @davidnuescheler

File: monitoring/homepage_test.js (hunk 2 above)
Comment (MarquiseRosier) [reply monitoring/homepage_test.js]: Nice! I included the statusMessage ^^ and logged the headers for more details!

File: monitoring/homepage_test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+function checkFeatured(err, response, body) {
+  const rgx = new RegExp('(?<=[\[\<])https://blog.adobe.com/en/(publish)?[0-9]{4}/[0-9]{2}/[0-9]{2}/.*?.html', 'g');
+  let match;
+  while ((match = rgx.exec(body)) !== null) {
+    $http.get(match[0]).on('response', detectFails(match[0]));
+  }
+}
+
+$http.get('https://blog.adobe.com/index.md', checkFeatured);
+$http.get('https://blog.adobe.com/en/query-index.json?limit=4&offset=0', checkSites);
```
Comment (rofe) [inline monitoring/homepage_test.js:48]: 4 may be too little though... I think we should make sure all of the first 13 cards on the homepage don't 404. Those typically include featured posts which are further back but still within the first 256. You could extract the featured posts from the markup.

**Review by trieloff — APPROVED** [review]

**Pull request merged**